servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/CancellableStack.java

```diff
@@ -36,8 +44,9 @@ public void cancel() {
      * or be cancelled immediately if this object's {@link #cancel()} method has already been called.
      * @param toAdd The {@link Cancellable} to add.
      * @return {@code true} if the {@code toAdd} was added. If {@code false} {@link Cancellable#cancel()} is called.
+     * @throws IllegalStateException if the maximum size would be exceeded by inserting this element.
      */
-    boolean add(Cancellable toAdd) {
+    boolean add(Cancellable toAdd) throws IllegalStateException {
```

Comment:
Javadoc should be enough, we try to use `throws` only for checked exceptions, like in Collection API.

Reply:
There is no harm in documenting runtime exceptions and it can improve IDE hints and static analysis.